feat(workflows): argo workflow chart, basic docs, fanout sample

dwmkerr opened the pull request.

dwmkerr committed:
feat(workflows): argo workflow chart, basic docs, fanout sample

dwmkerr committed:
feat: integrate argo-workflows chart with release pipeline

- Add argo-workflows chart to release-please config for version management
- Update chart version to 0.1.40 and ark-tenant dependency to 0.1.40
- Add argo-workflows to CI/CD build, release, and deploy workflows
- Improve query fanout evaluation prompt with structured output format
- Add workflow screenshot to documentation

dwmkerr committed:
fix: add helm dependency update before linting charts

Resolves chart linting error where dependencies (argo-workflows, ark-tenant)
were missing from charts/ directory

cm94242 commented on docs/content/developer-guide/workflows/argo-workflows.mdx:
````diff
@@ -0,0 +1,99 @@
+Install via Helm, or for local development mode use `devspace`:
+
+```bash
+helm upgrade --install argo-workflows \
+  oci://ghcr.io/mckinsey/agents-at-scale-ark/charts/argo-workflows
````
This doesn't seem to work.  Poking around on github maybe we still need to publish this, as i don't see it in packages

dwmkerr replied:
Yes my bad I shouldn've commented on this - this is the deployed chart location that'll only have the package in when we cut the build, I'll retest as soon as merged

cm94242 commented on docs/content/developer-guide/workflows/argo-workflows.mdx:
````diff
@@ -0,0 +1,99 @@
+
+When you get errors or unexpected results:
+
+```bash
+# List workflows
+kubectl get workflows -n argo-workflows
+
+# View logs
+argo logs @latest -n argo-workflows
+```
````
Users can probably connect the dots, but if we do devspace dev it seems like we're in the default namespace by default.

dwmkerr replied:
Miss on my side - have fixed and removed

cm94242 commented on docs/content/developer-guide/workflows/argo-workflows.mdx:
```diff
@@ -0,0 +1,99 @@
+
+# Or for local development, use devspace:
+cd services/argo-workflows
+devspace dev
```
was trying to play a bit with deployment options here with the -n option to push argo into its own namespace.    It seems that if you use default the first time, purge doesn't really clean the argo CRDs up, so subsequent ones fail.  Probably more of a curiousity than an actual problem, esp locally

dwmkerr replied:
<img width="826" height="329" alt="image" src="https://github.com/user-attachments/assets/7b7bd32c-0d7a-4b3c-a9a5-e4efe98cea84" />

good catch, added a note on this, we'll have to be more sophisticated with install instructions for multi-tenant (a bit like ark, there'll be 3x operators - argo, ark, minio operator)

dwmkerr committed:
docs: add CLI conventions for @latest and watch

- Add watch verb to common verbs list (not yet implemented)
- Document @latest alias convention following argo pattern
- Add @latest to parameter conventions table
- Add uninstallation instructions for argo-workflows including CRD cleanup

dwmkerr committed:
fix: correct QueryResponse type references in ark-dashboard

Changed QueryResponse-Output to QueryResponse across three files to match
the actual OpenAPI schema. The OpenAPI spec only defines QueryResponse, not
QueryResponse-Output.

- components/editors/evaluation-editor.tsx
- components/sections/queries-section.tsx
- lib/services/chat.ts

Also regenerated types and updated ark-sdk dependency path.

havasik committed:
feat: argo workflows cli tools sample (#382)

## Checklist

- [ ] Follow the [contributor guide](./CONTRIBUTING.md)
- [ ] End-to-end tests pass
- [ ] Unit tests for essential parts of your code, e.g. APIs exposed via
SDKs or endpoints
- [ ] Update `./docs`
- [ ] Recognize contributors with "@all-contributors please add <person>
for <code, bug, docs, etc>"
- [ ] Linked issues #eg101

dwmkerr committed:
docs: more sample info
